ui/app/src/components/BrokenReferencesDialog/BrokenReferencesDialogContainer.tsx

```diff
@@ -32,20 +31,52 @@ import DialogFooter from '../DialogFooter';
 import SecondaryButton from '../SecondaryButton';
 import PrimaryButton from '../PrimaryButton';
 import ApiResponseErrorState from '../ApiResponseErrorState';
+import { fetchDependant } from '../../services/dependencies';
+import type { LegacyItem } from '../../models';
+import { parseLegacyItemToContentItem } from '../../utils/content';
+import { pushDialog } from '../../state/actions/dialogStack';
 
 export function BrokenReferencesDialogContainer(props: BrokenReferencesDialogContainerProps) {
-	const { references, error, onClose, onContinue } = props;
+	const { path, references: initialReferences, error, onClose, onContinue } = props;
 	const dispatch = useDispatch();
 	const site = useActiveSiteId();
 	const { authoringBase } = useEnv();
+	const [references, setReferences] = useState(initialReferences || []);
 
 	const onContinueClick = (e) => {
 		onClose(e, null);
 		onContinue();
 	};
 
-	const onEditReferenceClick = (path: string) => {
-		dispatch(showEditDialog({ path, authoringBase, site, onSaveSuccess: fetchBrokenReferences() }));
+	const onEditReferenceClick = (referencePath: string) => {
+		dispatch(
+			pushDialog({
+				component: 'craftercms.components.LegacyFormDialog',
+
+				props: {
+					path: referencePath,
+					authoringBase,
+					site,
+					onSaveSuccess: () => {
+						// Fetch broken references again after editing
+						fetchDependant(site, path).subscribe({
+							next: (response: LegacyItem[]) => {
+								const refs = parseLegacyItemToContentItem(response);
+								setReferences(refs);
+							},
+							error: ({ response }) => {
+								dispatch(
+									pushDialog({
+										component: 'craftercms.components.ErrorDialog',
+										props: { error: response.response }
+									})
+								);
+							}
+						});
```

Comment:
Looks like before this just dispatched `fetchBrokenReferences`? Is it because references came directly from the store? If that's the case, what's that chunk of state used for?

Should it be removed or should we continue to use it but retrieve those internally, or...?